server/controllers/survey.ts

```diff
@@ -106,15 +115,25 @@ export function processMakeSurveyPage(req:Request, res: Response, next: NextFunc
  */
 export function processDeleteSurvey(req:Request, res: Response, next: NextFunction):void
 {
+    if (!req.user) { // if req.user is undefined
+        throw Error("Unreachable: this route handler is called only when the user is logged in");
+    }
+    const userId = req.user._id;
     const id = req.params.id;
-
-    Survey.findByIdAndRemove(id, {}, (err) => {
-        if(err)
-        {
+    // find the survey by survey id and user id
+    Survey.find({ owner:userId, _id:id},(err)=> {
+        if(err){
             return next(err);
         }
-        res.redirect("/account");
+        Survey.findByIdAndRemove(id, {}, (err) => {
```

Comment:
Same as above, you're requesting 2 database operations. What you want to use is this: [`Survey.findOneAndRemove`](https://mongoosejs.com/docs/api/model.html#model_Model.findOneAndRemove).

```ts
Survey.findOneAndRemove({ owner: userId, _id: id }, (err) => {
    if (err) {
        return next(err);
    }

    res.redirect("/account");
    // This is done regardless of if the survey was actually found with the given id and owner id.
    // In business applications, you might want to do separate things depending on that.
    // In our case, I think it's fine to leave it like this.
});
```